lib/plugs/interval_plug.ex

```diff
@@ -1,69 +1,39 @@
 defmodule Castle.Plugs.Interval do
   import Plug.Conn
 
-  @intervals %{
-    "1d"  => 86400,
-    "1h"  => 3600,
-    "15m" => 900,
-  }
-  @max_in_window 1000
-
   def init(default), do: default
 
   def call(conn, _default) do
     conn
-    |> set_interval()
+    |> assign(:interval, %{})
+    |> interval_part(:from, &Castle.Plugs.Interval.TimeFrom.parse/1)
+    |> interval_part(:to, &Castle.Plugs.Interval.TimeTo.parse/1)
+    |> interval_part(:seconds, &Castle.Plugs.Interval.Seconds.parse/1)
     |> round_time_window()
-    |> validate_window()
+    |> interval_struct()
```

Comment:
All this refactoring is basically to get the `Interval` struct instead of 3 separate params.  Tests are exercising the same functionality.